From f035e1e5abb19df5dced9c592ca986deac460435 Mon Sep 17 00:00:00 2001
From: bastila <20937049+silentfault@users.noreply.github.com>
Date: Thu, 23 Jul 2020 02:45:12 +0300
Subject: [PATCH] Fix overflow when window name is bigger than window width

---
 dwm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index 9fd0286..42cb8dd 100644
--- a/dwm.c
+++ b/dwm.c
@@ -931,7 +931,10 @@ drawbar(Monitor *m)
 					}
 					remainder--;
 				}
-				drw_text(drw, x, 0, tabw, bh, lrpad / 2, c->name, 0);
+				if (TEXTW(c->name) > w) /* title is bigger than the width of the title rectangle, don't center */
+					drw_text(drw, x, 0, tabw, bh, lrpad / 2, c->name, 0);
+				else /* center window title */
+					drw_text(drw, x, 0, tabw, bh, (tabw - TEXTW(c->name)) / 2, c->name, 0);
 				x += tabw;
 			}
 		} else {
2.27.0
