diff -up original/dwm.c patched/dwm.c
--- original/dwm.c	2025-09-22 17:19:55.369721948 +0200
+++ patched/dwm.c	2025-09-22 17:21:26.259383227 +0200
@@ -275,7 +275,9 @@ static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void zoom(const Arg *arg);
 static void load_xresources(void);
 static void resource_load(XrmDatabase db, char *name, enum resource_type rtype, void *dst);
-
+
+static void reload_xres(const Arg *arg);
+static void hidtoggle(const Arg *arg);
 static void focusmaster(const Arg *arg);
 
 /* variables */
@@ -1630,7 +1630,7 @@ setup(void)
 	wa.cursor = cursor[CurNormal]->cursor;
 	wa.event_mask = SubstructureRedirectMask|SubstructureNotifyMask
 		|ButtonPressMask|PointerMotionMask|EnterWindowMask
-		|LeaveWindowMask|StructureNotifyMask|PropertyChangeMask;
+		|LeaveWindowMask|StructureNotifyMask|PropertyChangeMask|KeyPressMask;
 	XChangeWindowAttributes(dpy, root, CWEventMask|CWCursor, &wa);
 	XSelectInput(dpy, root, wa.event_mask);
 	grabkeys();
@@ -2120,7 +2120,6 @@ spawn(const Arg *arg)
 {
 	struct sigaction sa;
 
-	if (arg->v == dmenucmd)
 		dmenumon[0] = '0' + selmon->num;
 	if (fork() == 0) {
 		if (dpy)
@@ -2709,7 +2709,34 @@ load_xresources(void)
 		resource_load(db, p->name, p->type, p->dst);
 	XCloseDisplay(display);
 }
-
+
+void reload_xres(const Arg* arg) {
+	load_xresources();
+}
+
+void
+hidtoggle(const Arg *arg)
+{
+	Client *c;
+	if (arg->v) { 
+		c = (Client*)arg->v;
+		if (HIDDEN(c))
+			showwin(c);
+		else {
+			hidewin(c);
+			focus(NULL);
+			arrange(c->mon);
+		}
+	} else {
+		c = selmon->sel;
+		if (!c) return;
+		if (selmon->hidsel)
+			show(NULL);
+		else
+			hide(NULL);
+	}
+}
+
 int
 main(int argc, char *argv[])
 {
